apps/system/utils/secure_log_cmd.c

```diff
@@ -0,0 +1,83 @@
+/****************************************************************************
+ *
+ * Copyright 2023 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <apps/shell/tash.h>
+#include <security/security_ss.h>
+
+/****************************************************************************
+ * Preprocessor Definitions
+ ****************************************************************************/
+#define LOG_SS_SLOT_PATH        "ss/32"
+
+/********************************************************************************
+ * Private Function Prototypes
+ ********************************************************************************/
+
+static void secure_log(int argc, char **args)
+{
+        security_handle hnd;
+	security_data input = {NULL, 0};
+	security_data output = {NULL, 0};
+
+	if (!strncmp(args[1], "on", 3)) {
+		input.data = "1";
+	} else if (!strncmp(args[1], "off", 4)) {
+		input.data = "0";
+	} else {
+		printf("Invalid argument\n");
+		return;
+	}
+	input.length = 1;
+
+        /* init */
+        if (0 != security_init(&hnd)) {
+		printf("Fail to init security handler\n");
+		return;
+	}
+
+        /* Write 'log on/off' variable to secure storage  */
+        if (0 != ss_write_secure_storage(hnd, LOG_SS_SLOT_PATH, 0, &input)) {
```

Comment:
1. in my opinion there's no reason divide security data to input & output
2. It should compare user input & changed value in ss level.
3. Below is my suggestion
```
int log_level;
security_data data = {NULL, 0};

if (!strncmp(args[1], "on", 3)) {
	data.data = "1";
	log_level = 1;
} else if (!strncmp(args[1], "off", 4)) {
	data.data = "0";
	log_level = 0;
} else {
	printf("Invalid argument\n");
	return;
}

...
ret = ss_write_secure_storage(... &data);
...
ret = ss_read_secure_storage(... &data);

if (log_level == 1) {
	if (!strncmp(output.data, "1", 2)) {
		printf("Assert log is enabled\n");
	} else {
		printf("error, set failed");
	}
} else {
	if (!strncmp(output.data, "0", 2)) {
		printf("Assert log is disabled\n");
	} else {
		printf("error, set failed");
	}
}
```